Vlang has temporarily switched to using a garbage collector.
This patch corrects the GC prefix.

Also adds macOS-specific flag check that will appear in next release.
https://github.com/vlang/v/commit/473d26ce472cb9d5d0696656e980545442eb32af

--- vlib/builtin/builtin_d_gcboehm.c.v.original	2022-06-25 13:00:27.000000000 +0100
+++ vlib/builtin/builtin_d_gcboehm.c.v	2022-06-25 13:01:46.000000000 +0100
@@ -31,12 +31,12 @@
 	#flag -DGC_BUILTIN_ATOMIC=1
 	$if macos || linux {
 		#flag -DGC_PTHREADS=1
-		#flag -I @VEXEROOT/thirdparty/libgc/include
-		$if (prod && !tinyc && !debug) || !(amd64 || arm64 || i386 || arm32) {
+		#flag -I @PREFIX@/include
+		$if (!macos && prod && !tinyc && !debug) || !(amd64 || arm64 || i386 || arm32) {
 			// TODO: replace the architecture check with a `!$exists("@VEXEROOT/thirdparty/tcc/lib/libgc.a")` comptime call
 			#flag @VEXEROOT/thirdparty/libgc/gc.o
 		} $else {
-			#flag @VEXEROOT/thirdparty/tcc/lib/libgc.a
+			#flag @PREFIX@/lib/libgc.a
 		}
 		#flag -ldl
 		#flag -lpthread
